scRNA-seq-advanced/04-differential_expression.Rmd

```diff
@@ -309,154 +309,230 @@ For this example we will focus on identifying DE genes in these `Tumor_Myoblast`
 
-```{r}
-# deseq2 requires the coldata object as a separate input
-#  in data.frame format
-coldata_df <- as.data.frame(colData(tumor_myoblast_sce))
-  
+```{r deseq object}
 # set up the deseq object 
-deseq_object <- DESeq2::DESeqDataSetFromMatrix(counts(tumor_myoblast_sce),
-                                               colData = coldata_df,
-                                               design = ~ diagnosis_group)
+deseq_object <- DESeq2::DESeqDataSet(tumor_myoblast_sce,
+                                     # formula for grouping samples 
+                                     design = ~ diagnosis_group)
 ```
 
-Use the median of ratios method for count normalization followed by regularized log transformation.
+If you look at the pseudo-bulked SCE object you should have only seen one assay, the `counts` assay.
+This is because `DESeq2` expects raw counts.
+When we run `DESeq2` on our dataset, raw counts will first be normalized using size factors to account for differences in total sample counts.
+Therefore we don't have to do any normalization on our own – we'll let `DESeq2` do all the work for us.
 
-```{r}
-# estimate size factors first 
+However, before we dive into DE analysis, we can do some initial exploration and visualization of our data to see if our samples separate by our known factor of interest, RMS subtype.
+In particular, we can use principal component analysis (PCA) of our pseudo-bulked dataset to visualize any variation between samples.
+If there is variation in their gene expression patterns we would expect samples from each RMS subtype to separate in PC space, likely indicating presence of differentially expressed genes.
+We can evaluate this by plotting PC1 and PC2.
+
+In order to create our PCA plot, we will first need to normalize our data to account for any technical variations across samples.
+As a reminder, this is NOT required for running `DESeq2` analysis, we are using it to visualize our data prior to DE analysis.
+
+```{r normalize}
+# estimate size factors first
 deseq_object <- DESeq2::estimateSizeFactors(deseq_object)
 
 # normalize and log transform to use for visualization
 normalized_object <- DESeq2::rlog(deseq_object, blind = TRUE, fitType = 'local')
 normalized_object
 ```
 
-Evaluate QC here using PCA. 
-Here we can label by the diagnosis groups to show that the highest amount of variance is due to different subdiagnoses. 
+We now have a normalized and transformed object that can be directly input to the `DESeq2::plotPCA()` function, which will both calculate and plot the PC results.
 
-```{r}
+```{r plotPCA}
 DESeq2::plotPCA(normalized_object, intgroup = "diagnosis_group")
 ```
 
-```{r}
+As expected we see that samples group together based on RMS subtype and are separated along the PC1 axis, the PC contributing the highest amount of variation.
+
+We'll now use the wrapper function `DESeq()` to perform our differential expression analysis.
```

Comment:
I think "wrapper" could be too jargon-y. We use "convenience function" for SingleR, does that work here?

```suggestion
We'll now use the convenience function `DESeq()` to perform our differential expression analysis.
```